### ISSUE TITLE
[E2503] fails incorrectly when intrinsic function used in Protocol value 

### ISSUE THREAD

**Issue by jaymccon** [awslabs/cfn-python-lint#272]
*cfn-lint version: 0.4.2*

*Description of issue.*

This is valid, and conforms to the spec, but rule throws an error:

```yaml
Parameters:
  TestParam:
    Type: String
    Default: TCP
Conditions:
  TestCond: !Equals ['a', 'a']
Resources:
  OpenShiftMasterELB:
    Type: AWS::ElasticLoadBalancing::LoadBalancer
    Properties:
      Subnets:
        - subnet-1234abcd
      SecurityGroups:
        - sg-1234abcd
      Listeners:
        # Fails on Protocol
        - InstancePort: '1'
          InstanceProtocol: !Ref TestParam
          LoadBalancerPort: '1'
          Protocol: !Ref TestParam
        # Also fails on Protocol
        - InstancePort: '2'
          InstanceProtocol: !If [TestCond, TCP, SSL]
          LoadBalancerPort: '2'
          Protocol: !If [TestCond, TCP, SSL]
        # Works
        - InstancePort: '3'
          InstanceProtocol: !If [TestCond, TCP, SSL]
          LoadBalancerPort: '3'
          Protocol: TCP
```

**Comment by cmmeyer** [comment]
Thanks! Catching all of the little intrinsic nuances seems to 90% of the game here.
